test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,341 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func resizeNodeInstanceGroup(size int) error {
+	// TODO: make this hit the compute API directly instread of shelling out to gcloud.
+	output, err := exec.Command("gcloud", "preview", "managed-instance-groups", "--project="+testContext.CloudConfig.ProjectID, "--zone="+testContext.CloudConfig.Zone,
+		"resize", testContext.CloudConfig.NodeInstanceGroup, fmt.Sprintf("--new-size=%v", size)).CombinedOutput()
+	if err != nil {
+		Logf("Failed to resize node instance group: %v", string(output))
+	}
+	return err
+}
+
+func nodeInstanceGroupSize() (int, error) {
+	// TODO: make this hit the compute API directly instread of shelling out to gcloud.
+	output, err := exec.Command("gcloud", "preview", "managed-instance-groups", "--project="+testContext.CloudConfig.ProjectID,
+		"--zone="+testContext.CloudConfig.Zone, "describe", testContext.CloudConfig.NodeInstanceGroup).CombinedOutput()
+	if err != nil {
+		return -1, err
+	}
+	pattern := "currentSize: "
+	i := bytes.Index(output, []byte(pattern))
+	if i == -1 {
+		return -1, fmt.Errorf("could not find '%s' in the output '%s'", pattern, output)
+	}
+	truncated := output[i+len(pattern):]
+	j := bytes.Index(truncated, []byte("\n"))
+	if j == -1 {
+		return -1, fmt.Errorf("could not find new line in the truncated output '%s'", truncated)
+	}
+
+	currentSize, err := strconv.Atoi(string(truncated[:j]))
+	if err != nil {
+		return -1, err
+	}
+	return currentSize, nil
+}
+
+func waitForNodeInstanceGroupSize(size int) error {
+	for start := time.Now(); time.Since(start) < 4*time.Minute; time.Sleep(20 * time.Second) {
```

Comment:
nit: maybe change the poll intervals (here and below) to 5 seconds? This could help the tests run faster...